ml-kem/src/lib.rs

```diff
@@ -132,10 +132,25 @@ pub trait KemCore {
     type CiphertextSize: ArraySize;
 
     /// A decapsulation key for this KEM
-    type DecapsulationKey: Decapsulate<Ciphertext<Self>, SharedKey<Self>> + Debug + PartialEq;
+    type DecapsulationKey: Decapsulate<Ciphertext<Self>, SharedKey<Self>>
+        + EncodedSizeUser
+        + Debug
+        + PartialEq;
 
+    #[cfg(not(feature = "deterministic"))]
     /// An encapsulation key for this KEM
-    type EncapsulationKey: Encapsulate<Ciphertext<Self>, SharedKey<Self>> + Debug + PartialEq;
+    type EncapsulationKey: Encapsulate<Ciphertext<Self>, SharedKey<Self>>
+        + EncodedSizeUser
+        + Debug
+        + PartialEq;
+
+    #[cfg(feature = "deterministic")]
+    /// An encapsulation key for this KEM
```

Comment:
Similar concern here
```suggestion
    /// An encapsulation key for this KEM
    #[cfg(feature = "deterministic")]
```